cmake/ProjectSettings.cmake

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# Set a default build type if none was specified
+
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui, ccmake
+    set_property(
+        CACHE CMAKE_BUILD_TYPE
+        PROPERTY STRINGS
+        "Debug"
+        "Release"
+        "MinSizeRel"
+        "RelWithDebInfo")
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Default install directory
+
+# If the user hasn't configured cmake with an explicit
+# CMAKE_INSTALL_PREFIX, then set it to safely install into ./dist, to
+# help prevent the user from accidentally writing over /usr/local or
+# whatever.
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND PROJECT_IS_TOP_LEVEL)
+    set(CMAKE_INSTALL_PREFIX
+        "${PROJECT_SOURCE_DIR}/dist" CACHE PATH "Installation location" FORCE)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Interprocedural Optimization
+
+option(ENABLE_IPO "Enable Interprocedural Optimization, aka Link Time Optimization (LTO)" OFF)
+
+if (ENABLE_IPO)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT result OUTPUT output)
+    if (result)
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+    else ()
+        message(SEND_ERROR "IPO is not supported: ${output}")
+    endif ()
+endif ()
+
+
+#-----------------------------------------------------------------------
+# IDE helpers
+
+# Generate compile_commands.json to make it easier to work with clang
+# based tools
+if (NOT DEFINED CMAKE_EXPORT_COMPILE_COMMANDS)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+endif ()
+
+# Use folders to organise targets in IDEs via the FOLDER target
+# property.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Colorize compiler diagnostics.
+if (CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
+    add_compile_options(-fcolor-diagnostics)
```

Comment:
Nit: Global options are a little frowned upon in modern CMake

Reply:
Good spot, will move to the per-target default properties function.

Reply:
Simply removed these options in 7d082bd2 since (as I put in the commit message)

> Remove explicitly forcing coloured diagnostics. Forcing to enabled
> could be messy in unsupported terminals (rendering control characters).
> 
> The defaults for compilers should hopefully do the right thing:
> * GCC: "default depends on how the compiler has been configured" - we
>   can just hope it's "auto", which "use[s] color only when the standard
>   error is a terminal".
> * Clang: "defaults to on when a color-capable terminal is detected".